lib/providers/update_provider.dart

```diff
@@ -0,0 +1,372 @@
+    final data = await downloads.read() as Map;
+
+    _automaticDownloads = data[kHiveAutomaticUpdates];
+    _lastCheck = DateTime.tryParse(data[kHiveLastCheck] ?? '');
+
+    if (notifyListeners) this.notifyListeners();
+  }
+
+  /// If there is anything loading at the time
+  bool loading = false;
+
+  /// Whether the user wants to automatically download updates
+  bool _automaticDownloads = false;
+  bool get automaticDownloads => _automaticDownloads;
+  set automaticDownloads(bool value) {
+    _automaticDownloads = value;
+    _save();
+  }
+
+  /// The last time the user checked for updates
+  DateTime? _lastCheck;
+  DateTime? get lastCheck => _lastCheck;
+  set lastCheck(DateTime? date) {
+    _lastCheck = date;
+    _save();
+  }
+
+  /// Checks how the Linux executable was installed.
+  ///
+  /// Returns `null` if the platform is not Linux or if the environment is not
+  /// recognized.
+  ///
+  /// The possible values are:
+  ///   * `rpm`
+  ///   * `deb`
+  ///   * `AppImage`
+  ///   * `tar.gz` (tarball)
+  ///
+  /// This means the value represent the file extension of the executable.
+  ///
+  /// Each Flutter executable is built with a different `linux_environment`
+  /// value, so it is possible to distinguish between them. This is useful to
+  /// know how to upgrade the app.
+  ///
+  /// See also:
+  ///
+  ///  * [install], which uses this method to install the correct executable.
+  String? get linuxEnvironment {
+    assert(Platform.isLinux);
+
+    if (!const bool.hasEnvironment('linux_environment')) return null;
+
+    return const String.fromEnvironment('linux_environment');
+  }
+
+  /// Check if updates are supported on the current platform.
+  ///
+  /// On Windows, updates are always supported.
+  ///
+  /// On Linux, updates are supported if the `linux_environment` is set and it
+  /// is not an `AppImage`.
+  bool get isUpdatingSupported {
+    if (Platform.isWindows) return true;
+    if (Platform.isLinux) {
+      return linuxEnvironment != null &&
+          linuxEnvironment != UpdateVersion.appImage;
+    }
+
+    return false;
+  }
+
+  /// Whether any executable is being downloaded at the time
+  bool downloading = false;
+
+  /// The progress of the download, from 0.0 to 1.0
+  double downloadProgress = 0.0;
+
+  /// Gets the executable for the given [version].
+  ///
+  /// If the executable is not found, returns `null`. To download the executable,
+  /// call [download(version)].
+  File? executableFor(String version) {
+    assert(isDesktop, 'This should never be reached on non-desktop platforms');
+
+    if (Platform.isWindows) {
+      const fileName = 'bluecherry-windows-setup';
+      final file = File(path.join(tempDir, '$fileName-$version.exe'));
+      if (file.existsSync()) return file;
+    } else if (Platform.isLinux) {
+      final fileName = 'bluecherry-linux-x86_64-$version.$linuxEnvironment';
+      final file = File(fileName);
```

Comment:
Temporary dir is not taken into account here. Also, I'd rather make use of `path.setExtension`